executor/src/witgen/jit/function_cache.rs

```diff
@@ -212,25 +209,45 @@ impl<'a, T: FieldElement> FunctionCache<'a, T> {
             );
         }
 
-        log::trace!("Generated code ({} steps)", code.len());
+        log::trace!("Generated code ({} steps)", processor_result.code.len());
+        Some((processor_result, prover_functions))
+    }
+
+    fn compile_witgen_function(
+        &self,
+        result: ProcessorResult<T>,
+        prover_functions: Vec<ProverFunction<'a, T>>,
+        cache_key: &CacheKey<T>,
+    ) -> Option<CacheEntry<T>> {
         let known_inputs = cache_key
             .known_args
             .iter()
             .enumerate()
             .filter_map(|(i, b)| if b { Some(Variable::Param(i)) } else { None })
             .collect::<Vec<_>>();
 
+        let has_prover_function_call = has_prover_function_call(&result.code);
+        // Use the compiler for goldilocks with at most MAX_COMPILED_CODE_SIZE statements and
+        // the interpreter otherwise.
+        let interpreted = matches!(T::known_field(), Some(KnownField::GoldilocksField))
+            || (code_size(&result.code) > MAX_COMPILED_CODE_SIZE && !has_prover_function_call);
```

Comment:
`has_prover_function_call` should not be tested here due to the following `if`

Reply:
I changed it again. The idea is that this test is where we want to get to.